index.ts

```diff
@@ -42,39 +28,55 @@ async function indexSignaturesForAddress(
   until?: TransactionSignature
 ) {
   let sigs: ConfirmedSignatureInfo[];
+  let latest = before;
+  let earliest = until;
   do {
     sigs = await rpc.getSignaturesForAddressWithRetries(address, {
-      before,
-      until,
+      before: latest,
+      until: earliest,
       limit: 1000,
     });
     sigs.reverse();
+    // Edge case where you enter indexing but there's nothing to index
+    if (!sigs || sigs.length === 0) {
+      // Edge case when before is undefined and there are no new txes to pull from rpc
+      return { earliest: "", latest: before ? before : until };
+    }
+
+    latest = sigs[sigs.length - 1].signature;
+    earliest = sigs[0].signature;
```

Comment:
Are we sure this the the right way around for earliest and latest? On postman the last element in the returned array is the earliest tx (earliest blocktime)

Reply:
Yeah so fetching from RPC you get it in reverse order (zero index is the most recent/latest). If you look a few lines above I reverse the list that's returned.

Reply:
But yes if you have better terminology for earliest/latest lmk, I thought the before/until stuff in the RPC args was even more confusing. I tend to refer to earliest being earliest in absolute time of when the tx was executed on the blockchain.